### ISSUE TITLE
Torch autograd expects Variable with requires_grad set to True

### ISSUE THREAD

**Issue by ags3927** [roman-vygon/triplet_loss_kws#6]
Torch autograd expects Variable with requires_grad set to True but doesn't find such Variables. Any idea as to how to get around this?

**Comment by ToughmanL** [comment]
In pytorch, you need set loss.requres_grad = True.
In this project, loss fuction locate in triplet.py:line 50.
I solved it through the following steps：
        mean_losses = losses.mean()
        mean_losses.requires_grad=True
        return mean_losses

**Comment by roman-vygon** [comment]
Please see if uncommenting [this](https://github.com/roman-vygon/triplet_loss_kws/blob/master/layers/l2.py#L36) line helps, or the @ToughmanL's answer.
